### ISSUE TITLE
unexpected training loss and accuracy when training with cldice

### ISSUE THREAD

**Issue by aryaabdi** [jocpae/clDice#33]
Hi, I tried using this loss function for a binary segmentation task and compared the results to DiceCELoss. However, the training loss and accuracy behaved unexpectedly—loss drops to zero early in training, while accuracy suggests divergence. Please see the TensorBoard snapshot here: [the grey curves]

![Image](https://github.com/user-attachments/assets/416a49a7-65fc-4eb6-bf4c-ba95155f3673)

Any suggestions what may be wrong?
